### ISSUE TITLE
Errors thrown by the hover provider are not shown to the user in the hover

### ISSUE THREAD

**Issue by felixfbecker** [sourcegraph/sourcegraph#811]
It used to be that if a language server returned an error response, that error is shown in the hover. Codeintellify has code and styles around this.

Now, if a hover provider throws, the error is logged but not shown anywhere. This means the user has no idea what is going on and language servers / extensions have no way to report errors back to the user for a specific request in the context of the token.

![2018-11-06 17 44 36](https://user-images.githubusercontent.com/10532611/48079561-e62be280-e1eb-11e8-8c51-142f12fa5209.gif)

**Comment by sqs** [comment]
We should show hover errors in the hover. If there are multiple hover providers, then the successful hover responses should be shown as normal, and there should be red bg sections in the hover with the error message. 

I believe VS Code suppresses errors from hover providers entirely like we do here, but that seems to make debugging harder. I can see why they did that, but I think we can take a harder line against extensions not catching exceptions in their hover handler. 

As an immediate workaround, extensions can call showMessage to show an error to users.

**Comment by chrismwendt** [comment]
For context: I had [changed this behavior](https://github.com/sourcegraph/sourcegraph-langserver-http/pull/8) primarily to avoid showing errors on every single hover in the specific case where the user is viewing a private repository that did not exist on the Sourcegraph instance. That was a heavy-handed solution. A more UX-friendly approach could be to choose where errors show up based on the error.

**Comment by felixfbecker** [comment]
In that PR it says

> Errors returned by the language server will still show up in the hover tooltips.

which doesn't seem true anymore?
Also, that PR is to the langservers-http extension, but I observed this behaviour with my own extension, so it may not be related?

Imo errors should always be shown. If an error should not be shown, the provider shouldn't return one.

**Comment by chrismwendt** [comment]
This could be related to https://github.com/sourcegraph/sourcegraph-extension-api/pull/117

**Comment by felixfbecker** [comment]
Yes, that look like the cause. Not showing any errors is not a good fix, instead errors should be aggregated like results are aggregated. @attfarhan

**Comment by felixfbecker** [comment]
This makes debugging the new language servers a lot harder because there are a lot of logs in the console and it's easy to miss errors and hard to map them to requests

**Comment by sqs** [comment]
I'll fix tonight
